### ISSUE TITLE
Enhancing Godot Engine :)

### ISSUE THREAD

**Issue by RebelliousX** [godotengine/godot#3161]
- Add a viewer to Project's root node hierarchical relationship to all referenced nodes in the project.
When the project grows, finding the relationship between scenes becomes awkward and hard to maintain.
By adding a Project mode, "Scene" like tab to show heirarchy of the object (a ListTree would suffice).
It would make working with Godot more robust than ever.

In the Instancing wiki tutorial: see the following image.
https://github.com/godotengine/godot/wiki/images/shooter_instancing.png

If Godot can show this kind of relationship between scenes it would be awesome.

I am coming from Game Maker Studio Pro, and I really hate to compare it with Godot since both
are excellent, and both have different ways of doing the management. But at least Game Maker Studio
does have excellent project viewer. It even creates logical filter "Folders" for resources or Rooms (Scenes) in the Project's ListTree.

Please consider adding this feature.

- When renaming any kind of object in Godot, it should find all occurrences in scripts/editor
and change that as well. This can be achieved by parsing script files, any matching names for the changed 
scene/node, like "load(), preload(), set_script() get_node()..etc!) functions.

- There is an awesome library called "stb_image" https://github.com/nothings/stb
It is a single header library, no license at all, can be used without any creditation.
Basically they are Public Domain, it has great number of file formats.

There are other header files too in that project to open OggVorbis, TrueType fonts...etc
also public domain.

- Please consider adding more image formats, a vectorized SWF file would be awesome.
To avoid pixelated images while zooming.

- I know I am stretching it, but it would be great if the users would be able to create a
breakpoint by clicking  the area left to line numbers by making it a few pixels wider 
(similar to Visual Studio), also adding a context menu to create custom breakpoint 
that occurs when a condition is true, for example (only do a breakpoint `if x value > 30`)

Thanks for the great work on Godot, you guys are awesome.

**Comment by akien-mga** [comment]
Please avoid listing many unrelated things in the same issue, each bug/feature proposal should have its own issue so that it can be assessed and worked on separately.

I'll keep this one open as there are probably interesting ideas, but be aware that some points might be missed by developers when they skim through the topic in 30 s :)

**Comment by kubecz3k** [comment]
It would be great @RebelliousX if you could open new issue for each idea,  we could discuss each one of them  in a separate thread and it would be a lot easier to decide what to implement and when in the situation when we could thread them as separate ideas.

**Comment by RebelliousX** [comment]
Oh my bad, I will. Sorry about that.

**Comment by RebelliousX** [comment]
I am closing the issue, I added separate issues for each suggestion.

**Comment by akien-mga** [comment]
Thanks!
